test/e2e/main.go

```diff
@@ -0,0 +1,811 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	operatorclient "github.com/coreos-inc/tectonic-operators/operator-client/pkg/client"
+	"github.com/kubernetes-incubator/apiserver-builder/pkg/controller"
+	appsv1beta2 "k8s.io/api/apps/v1beta2"
+	apiv1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	apiextensionsclientset "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	apiregistrationv1beta1 "k8s.io/kube-aggregator/pkg/apis/apiregistration/v1beta1"
+	"sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset"
+)
+
+const (
+	pollInterval                            = 1 * time.Second
+	timeoutPoolInterval                     = 5 * time.Second
+	timeoutPoolAWSInterval                  = 25 * time.Second
+	timeoutPoolClusterAPIDeploymentInterval = 10 * time.Minute
+	timeoutPoolMachineSetRunningInterval    = 10 * time.Minute
+
+	defaultLogLevel          = "info"
+	targetNamespace          = "default"
+	awsCredentialsSecretName = "aws-credentials-secret"
+	region                   = "us-east-1"
+	machineSetReplicas       = 2
+)
+
+func usage() {
+	fmt.Printf("Usage: %s\n\n", os.Args[0])
+}
+
+// TestConfig stores clients for managing various resources
+type TestConfig struct {
+	CAPIClient          *clientset.Clientset
+	APIExtensionsClient *apiextensionsclientset.Clientset
+	KubeClient          *kubernetes.Clientset
+	OperatorClient      operatorclient.Interface
+	AWSClient           *AWSClient
+}
+
+// NewTestConfig creates new test config with clients
+func NewTestConfig(kubeconfig string) *TestConfig {
+	config, err := controller.GetConfig(kubeconfig)
+	if err != nil {
+		glog.Fatalf("Could not create Config for talking to the apiserver: %v", err)
+	}
+
+	capiclient, err := clientset.NewForConfig(config)
+	if err != nil {
+		glog.Fatalf("Could not create client for talking to the apiserver: %v", err)
+	}
+
+	apiextensionsclient, err := apiextensionsclientset.NewForConfig(config)
+	if err != nil {
+		glog.Fatalf("Could not create client for talking to the apiserver: %v", err)
+	}
+
+	kubeClient, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		glog.Fatalf("Could not create kubernetes client to talk to the apiserver: %v", err)
+	}
+
+	opclient := operatorclient.NewClient(kubeconfig)
```

Comment:
No need to this client. Which means the `github.com/coreos-inc/tectonic-operators/operator-client/pkg/client` dependency can be removed from the vendor tree.